### PR TITLE
Move `sample-dataset` into `test-data` for backend tests

### LINKED ISSUE #29573: Combine `sample-dataset` and `test-data` data in backend tests
Roll the `sample-dataset` tables into the `test-data` definition, so they're the same database. This will make things a little more convenient when writing tests and increase performance slightly in tests

- [ ] Move definitions from `sample-dataset.edn` into `test-data.edn`; 
- [ ] Remove all the usages of `(mt/dataset sample-dataset ...)` in tests
- [ ] Cycle the `v3_` key for BigQuery (if we're still prefixing it that way; we may change it in the near future) or any other Cloud databases that cache test dataset data loading. Athena will have to be done manually from a REPL. 
- [ ] We will probably need to update a few tests

### PR DESCRIPTION
**Pull request by kienstra**
Closes https://github.com/metabase/metabase/issues/29573

### Description
In the backend tests, moves `sample-dataset` into `test-data`, and updates tests

|  | Task | What I did | 
|---|---|---|
 | ✅ | Move definitions from `sample-dataset.edn` into `test-data.edn` | Moved [sample-dataset.edn](https://github.com/kienstra/metabase/blob/373f4c4ff27966fa913feb1dfea89b75a0c27425/test/metabase/test/data/dataset_definitions/sample-dataset.edn) verbatim into the end of [test-data.edn](https://github.com/kienstra/metabase/blob/2e2fd689f162e3656bd5aa6c6548fd761823da9e/test/metabase/test/data/dataset_definitions/test-data.edn) |  
 | ✅ |  Remove all the usages of `(mt/dataset sample-dataset ...)` in tests | Removed all of these, and only replaced them with`(mt/dataset test-data` in 3 places where they [were](https://github.com/kienstra/metabase/blame/921169e35ea6d9a82959eae5114d2ec7c0949492/modules/drivers/athena/test/metabase/test/data/athena.clj#L93) [needed](https://github.com/kienstra/metabase/blob/296f1dee8ed617d6c3f258643f5e9f4bed7a656b/test/metabase/api/action_test.clj#L197) | 
 | ✅ | Cycle the `v3_` key for BigQuery (if we're still prefixing it that way; we may change it in the near future) or any other Cloud databases that cache test dataset data loading. Athena will have to be done manually from a REPL.  |  Cycled the [BigQuery](https://github.com/kienstra/metabase/blame/296f1dee8ed617d6c3f258643f5e9f4bed7a656b/modules/drivers/bigquery-cloud-sdk/test/metabase/test/data/bigquery_cloud_sdk.clj#L57) and [Snowflake](https://github.com/kienstra/metabase/blame/296f1dee8ed617d6c3f258643f5e9f4bed7a656b/modules/drivers/snowflake/test/metabase/driver/snowflake_test.clj#L48) keys from `v3_` to `v4_`. Didn't update Athena.  |  
 | ✅ | We will probably need to update a few tests |  Updated tests so all the tests running in [my fork](https://github.com/kienstra/metabase/pull/6) pass ([example](https://github.com/kienstra/metabase/blame/296f1dee8ed617d6c3f258643f5e9f4bed7a656b/test/metabase/api/database_test.clj#L670-L682)). Other tests that aren't running in my fork might fail if I missed something, [like BigQuery](https://github.com/metabase/metabase/blob/d149840374c9e020ccb206271f4a14da8a371f45/.github/workflows/drivers.yml#L72-L94). |  

(The "Task" column is copy-pasted from Cam Saul's https://github.com/metabase/metabase/issues/29573)
### How to verify

GitHub actions should all pass.

[My fork](https://github.com/kienstra/metabase/pull/6) is running some of those actions. 

### Checklist

- [x] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

File: dev/test/dev/debug_qp_test.clj [hunk 1]
```diff
@@ -28,56 +28,55 @@
...
-             (debug-qp/expand-symbolize [:field "wow" {:base-type :type/Text, :source-field (mt/id :orders :product_id)}]))))
-    (testing "Binned field - no expansion (%id only)"
-      (is (= [:field '%people.source {:binning {:strategy :default}}]
-             (debug-qp/expand-symbolize [:field (mt/id :people :source) {:binning {:strategy :default}}]))))
-    (testing "Field with temporal unit"
-      (is (= '!default.created_at
-             (debug-qp/expand-symbolize [:field (mt/id :orders :created_at) {:temporal-unit :default}]))))
-    (testing "Field with join alias AND temporal unit"
-      (is (= '!default.&P1.created_at
-             (debug-qp/expand-symbolize [:field (mt/id :orders :created_at) {:temporal-unit :default, :join-alias "P1"}]))))
```
Comment (kienstra) [inline dev/test/dev/debug_qp_test.clj:62]: Most of this PR's diff is whitespace-only, from removing `(mt/dataset sample-dataset` and unindenting the lines it wrapped.

You're probably well aware of this, but it helped me to look at this with [whitespace disabled](https://github.com/metabase/metabase/pull/35973/files?diff=unified&w=1).

File: enterprise/backend/test/metabase_enterprise/advanced_permissions/models/permissions_test.clj [hunk 2]
```diff
@@ -50,40 +50,90 @@
           (is (nil? (download-perms-by-group-id group-id))))
 
         (testing "Download perms for individual tables can be set and revoked"
-          (let [[id-1 id-2 id-3 id-4] (map u/the-id (database/tables (mt/db)))]
+          (let [[id-1 id-2 id-3 id-4 id-5 id-6 id-7 id-8] (map u/the-id (database/tables (mt/db)))]
```
Comment (kienstra) [inline enterprise/backend/test/metabase_enterprise/advanced_permissions/models/permissions_test.clj:53]: Now that `sample-dataset` is moved into `test-data`, there are 8 ids (tables), not 4.

File: enterprise/backend/test/metabase_enterprise/advanced_permissions/query_processor/middleware/permissions_test.clj [hunk 3]
```diff
@@ -187,10 +187,14 @@
...
+      (with-download-perms (mt/id) {:schemas {"PUBLIC" {(mt/id 'users)      :full
+                                                        (mt/id 'categories) :full
+                                                        (mt/id 'venues)     :limited
                                                         (mt/id 'checkins)   :full
-                                                        (mt/id 'users)      :full
-                                                        (mt/id 'categories) :full}}}
+                                                        (mt/id 'products)   :limited
+                                                        (mt/id 'people)     :limited
+                                                        (mt/id 'reviews)    :limited
+                                                        (mt/id 'orders)     :limited}}}
```
Comment (kienstra) [inline enterprise/backend/test/metabase_enterprise/advanced_permissions/query_processor/middleware/permissions_test.clj:197]: This now has the tables from `sample-dataset`, like `orders`

File: dev/test/dev/debug_qp_test.clj [hunk 4]
```diff
@@ -28,56 +28,55 @@
               :fk-field-id  (mt/id :venues :category_id)}])))))
 
 (deftest to-mbql-shorthand-test
-  (mt/dataset sample-dataset
-    (testing "Normal Field ID clause"
-      (is (= '$user_id
-             (debug-qp/expand-symbolize [:field (mt/id :orders :user_id) nil])))
-      (is (= '$products.id
-             (debug-qp/expand-symbolize [:field (mt/id :products :id) nil]))))
```
Comment (kienstra) [inline dev/test/dev/debug_qp_test.clj:36]: Most of this PR's diff is whitespace-only, from removing `(mt/dataset sample-dataset` and unindenting the lines it wrapped.

You're probably well aware of this, but it helped me to look at this with [whitespace disabled](https://github.com/metabase/metabase/pull/35973/files?diff=unified&w=1).

File: test/metabase/driver/sql_jdbc_test.clj [hunk 5]
```diff
@@ -17,7 +17,7 @@
 (set! *warn-on-reflection* true)
 
 (deftest ^:parallel describe-database-test
-  (is (= {:tables (set (for [table ["CATEGORIES" "VENUES" "CHECKINS" "USERS"]]
+  (is (= {:tables (set (for [table ["CATEGORIES" "VENUES" "CHECKINS" "USERS" "ORDERS" "PEOPLE" "PRODUCTS" "REVIEWS"]]
```
Comment (kienstra) [inline test/metabase/driver/sql_jdbc_test.clj:20]: Now that `sample-dataset` is moved into `test-data`, `test-data` has 8 tables, not 4.

File: test_resources/load-durations.edn [hunk 6]
```diff
@@ -33,6 +29,10 @@
  ["test-data" "checkins"]                                             130413900
  ["test-data" "users"]                                                23762400
  ["test-data" "venues"]                                               29380500
+ ["test-data" "orders"]                                               3439047900
+ ["test-data" "people"]                                               1922302400
+ ["test-data" "products"]                                             95636200
+ ["test-data" "reviews"]                                              292738400
```
Comment (kienstra) [inline test_resources/load-durations.edn:35]: These load durations aren't right anymore, as they are from when they were in the `sample-dataset`. 

But I moved them to `test-data` for now.

**Comment by kienstra** [comment]
Must revert https://github.com/metabase/metabase/pull/35973/commits/d4c8f129461f7f2f5a06d3706648909abc013391 before considering a merge.

**Review by snoe — APPROVED** [review]
This is a great change, thanks for the contribution!

**Comment by kienstra** [comment]
Case, thanks for your huge amount of patience in https://github.com/metabase/metabase/pull/36064, and for making the Athena tests pass.

**Comment by kienstra** [comment]
Would it help if I clicked 'Update branch' to pull in `master`? Thought you have permissions to do that.

**Comment by snoe** [comment]
No, we'll take it from here.

**Pull request merged**